Library/Homebrew/utils/fork.rb

```diff
@@ -64,21 +67,20 @@ def self.safe_fork(&_block)
             socket.close
           end
           write.close
-          # Each line on the error pipe contains a JSON-serialized exception.
-          # We read the first, since only one is necessary for a failure.
-          data = read.gets
+          data = read.read
           read.close
           Process.wait(pid) unless socket.nil?
 
+          raise Interrupt if $CHILD_STATUS.exitstatus == 130
```

Comment:
Could you add a comment explaining why the `130` here? I notice that it's code that's moved from below but if you have context now it'd be great.